### PR TITLE
Add AuthParam class

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
Add an AuthScheme class that allows specifying name, value (dynamically) and rbsType.  
Remove special casing of operation_name parameter.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/projections/high_score_service/lib/high_score_service/config.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ module HighScoreService
   #   @option args [Hearth::HTTP::Client] :http_client (Hearth::HTTP::Client.new)
   #     The HTTP Client to use for request transport.
   #   @option args [Hearth::IdentityResolver] :http_login_identity_resolver
-  #     A Hearth::IdentityResolver that returns a Hearth::Identities::HTTPLogin for operations modeled with the smithy.api#httpBasicAuth auth scheme.
+  #     A Hearth::IdentityResolver that returns a Hearth::Identities::HTTPLogin for operations modeled with the smithy.api#httpDigestAuth auth scheme.
```
Comment (alextwoods) [inline codegen/projections/high_score_service/lib/high_score_service/config.rb:34]: http_login_identity_resolver. (HttpLogin)... is used for both smithy.api#httpBasicAuth and smithy.api#httpDigestAuth right?

When both are configured, this leads to maybe confusing documentation here.

Comment (mullermp) [reply codegen/projections/high_score_service/lib/high_score_service/config.rb:34]: Yep it should say both

Comment (alextwoods) [reply codegen/projections/high_score_service/lib/high_score_service/config.rb:34]: I dont really see a good way around this though other than updating this documentation to be more generic (trying to dynamically combine them.... its tricky and special cased).

Comment (mullermp) [reply codegen/projections/high_score_service/lib/high_score_service/config.rb:34]: Generic is fine .. "operations modeled to use it"

**Review by mullermp — APPROVED** [review]

**Pull request merged**